lib/client/conntest/ssh.go

```diff
@@ -17,24 +17,38 @@ limitations under the License.
 package conntest
 
 import (
+	"bytes"
 	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
 
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/defaults"
 	"github.com/gravitational/teleport/api/types"
-	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/api/utils/sshutils"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/client"
+	libsshutils "github.com/gravitational/teleport/lib/sshutils"
 )
 
 // SSHConnectionTester implements the ConnectionTester interface for Testing SSH access
 type SSHConnectionTester struct {
-	cd services.ConnectionsDiagnostic
+	userClt  auth.ClientI
+	proxyClt auth.ClientI
```

Comment:
Is one of these a client with user's identity and the other with proxy's identity?

Reply:
Yeah
I had to bring the proxy client to be able to fetch the Node's information
Otherwise, we would depend on the User's identity having `read` access to `node` resource.